### DISCUSSION TITLE
Add `with_args` to external_program to add arguments

### DISCUSSION THREAD

**Discussion by virtuald** [mesonbuild/meson#13667] [Ideas]
I'm specifically thinking this would be useful for executing python modules (`python -m foo` -- on Windows installed scripts aren't often available on the PATH), but it's general enough that it could save typing and aid readability in other places too.

Currently if you want to not repeat yourself a bunch, you need to do something like this:

```meson
py = pymod.find_installation()
mod_args = ['-m', 'mymod']

...

tgt = custom_target(... command: [py] + mod_args + ['@INPUT@', '@OUTPUT@'])
```

I propose making the following possible for any external program:

```meson
py = pymod.find_installation()
mymod = py.with_args('-m', 'mymod')

tgt = custom_target(... command: [mymod, '@INPUT@', '@OUTPUT@'])

```

The `with_args` function would return a new external_program object (without changing the original), and would add to the internal command list. In theory, all existing usages of external_program should continue to work, as the internal command is already a list due to automatic detection of shebangs and etc.

Open to bikeshedding on the method name.

**Comment by eli-schwartz** [comment]
Why don't you just do:
```python
py = pymod.find_installation()
mymod = [py, '-m', 'mymod']

tgt = custom_target(... command: [mymod, '@INPUT@', '@OUTPUT@'])
```

**Reply by eli-schwartz** [reply]
Note that the `command: ` kwarg, like many kwargs, knows that it is not semantically meaningful to take a recursive array such as `[[X, Y], Z]` and try to pass it as an array of arrays to e.g. subprocess command lines. Instead, meson flattens the arrays, so that it gets interpreted as `[X, Y, Z]`, when translating it into the custom_target command and writing out a backend rule.

**Reply by virtuald** [reply]
I didn't realize that was possible, neat. Does that work everywhere that external_program is supported?
